runtime/doc/filetype.txt

```diff
@@ -663,6 +663,18 @@ To disable this behavior, set the following variable in your vimrc: >
 
 	let g:python_recommended_style = 0
 
+QUERY					                *ft-query-plugin*
+
+
+By default enables lints tree-sitter query for installed parsers using
+|lua-treesitter-query_linter|. Query linting can be disabled using >
+
+	:let g:query_disable_linter = 1
+<
+The events that trigger linting are by default `BufWrite` and `CursorHold`.
```

Comment:
`InsertLeave` causes considerable lag. Please feel free to suggest any autocmd or cmd default configuration

Reply:
This file is almost sorted alphabetically, almost....

Reply:
How is `CursorHold` better than `InsertLeave` when the command is laggy?

Reply:
I'd remove `CursorHold`. Either use `BufWrite` or `TextChanged`

Reply:
The only costly check is loading the queries. Maybe just leave the cheap checks for the first iteration. Any checks that block typing (i.e. InsertLeave, TextChanged) are not possible with the current linter performance now. We could just have `BufWrite`

Reply:
(`InsertLeave` should not block typing; that's the whole point...)